stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocketServiceImpl.kt

```diff
@@ -42,7 +42,11 @@ internal class ChatSocketServiceImpl constructor(
     private val eventUiHandler = Handler(Looper.getMainLooper())
     private val healthMonitor = HealthMonitor(
         object : HealthMonitor.HealthCallback {
-            override fun reconnect() = this@ChatSocketServiceImpl.reconnect(connectionConf)
+            override fun reconnect() {
+                if (state is State.DisconnectedTemporarily) {
```

Comment:
`HealthMonitor` should only retry to reconnect if the current state is `DisconnectedTemporarily`, in another case it shouldn't